param/parameterized.py

```diff
@@ -151,6 +151,7 @@ def discard_events(parameterized):
         parameterized.param._events = events
 
 
+@lru_cache(maxsize=128)
```

Comment:
lru_cache wasn't [introduced until Python 3.2](https://docs.python.org/3/library/functools.html#functools.lru_cache). and I don't think Param should abandon Python 2.7 until Param version 2.0. Maybe define it as a no-op for 2.7?

Reply:
Removed the LRU cache for now but may end up readding it for something else later.